src/transaction.js

```diff
@@ -198,20 +198,23 @@ Transaction.prototype.hashForSignature = function(inIndex, prevOutScript, hashTy
   }
 
   typeForce('Number', inIndex)
-  typeForce('Script', prevOutScript)
+  if(prevOutScript)
+    typeForce('Script', prevOutScript)
   typeForce('Number', hashType)
 
-  assert(inIndex >= 0, 'Invalid vin index')
+  if(prevOutScript)
+    assert(inIndex >= 0, 'Invalid vin index')
   assert(inIndex < this.ins.length, 'Invalid vin index')
 
   var txTmp = this.clone()
-  var hashScript = prevOutScript.without(opcodes.OP_CODESEPARATOR)
+  var hashScript = prevOutScript ? prevOutScript.without(opcodes.OP_CODESEPARATOR) : null
 
   // Blank out other inputs' signatures
   txTmp.ins.forEach(function(txIn) {
     txIn.script = Script.EMPTY
   })
-  txTmp.ins[inIndex].script = hashScript
+  if(hashScript != null && inIndex >= 0)
+    txTmp.ins[inIndex].script = hashScript
```

Comment:
Nit: please adhere to project style by adding whitespace after the `if` keyword, and always use strict equality where possible.

``` javascript
if (hashScript !== null && inIndex >= 0) {
  txTmp.ins[inIndex].script = hashScript
}
```